protocol/test/match.ts

```diff
@@ -0,0 +1,220 @@
+import { expect } from "chai";
+import { ethers } from "hardhat";
```

Comment:
here can be seen that `ethers` is imported from `hardhat`, which constructs an instance of `ethers` with all relevant testing infrastructure